src/compute-types/src/dataflows.rs

```diff
@@ -439,26 +439,90 @@ where
     }
 }
 
-impl<P: PartialEq, S: PartialEq, T: timely::PartialOrder> DataflowDescription<P, S, T> {
+impl<P, S, T> DataflowDescription<P, S, T>
+where
+    P: PartialEq,
+    S: Clone + PartialEq,
+    T: Clone + timely::PartialOrder,
+{
     /// Determine if a dataflow description is compatible with this dataflow description.
     ///
     /// Compatible dataflows have equal exports, imports, and objects to build. The `as_of` of
     /// the receiver has to be less equal the `other` `as_of`.
     ///
     // TODO: The semantics of this function are only useful for command reconciliation at the moment.
     pub fn compatible_with(&self, other: &Self) -> bool {
```

Comment:
Nit: Update documentation to mention that this call normalizes transient IDs, and that its cost is proportional to `self` and `other`.